subsys/bluetooth/host/hci_core.c

```diff
@@ -1042,6 +1042,36 @@ static void le_conn_complete_adv_timeout(void)
 }
 
 static void enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
+{
+#if (CONFIG_BT_ID_MAX > 1) && (CONFIG_BT_EXT_ADV_MAX_ADV_SET > 1)
+	if (IS_ENABLED(CONFIG_BT_PERIPHERAL) &&
+		evt->role == BT_HCI_ROLE_SLAVE &&
+		evt->status == 0 &&
+		(IS_ENABLED(CONFIG_BT_EXT_ADV) &&
+				BT_FEAT_LE_EXT_ADV(bt_dev.le.features))) {
+
+		/* Cache the connection complete event. Process it later.
+		 * See bt_dev.cached_conn_complete.
+		 */
+		for (uint8_t i = 0; i < ARRAY_SIZE(bt_dev.cached_conn_complete); i++) {
+			if (!bt_dev.cached_conn_complete[i].valid) {
+				memcpy(&bt_dev.cached_conn_complete[i].evt,
+					evt,
+					sizeof(struct bt_hci_evt_le_enh_conn_complete));
+				bt_dev.cached_conn_complete[i].valid = true;
+				return;
+			}
+		}
+
+		BT_WARN("No more cache entries available. This should not happen by design");
+
+		return;
+	}
+#endif
+	bt_hci_le_enh_conn_complete(evt);
+}
```

Comment:
```suggestion
	}
#else 
	bt_hci_le_enh_conn_complete(evt);
#endif /* (CONFIG_BT_ID_MAX > 1) && (CONFIG_BT_EXT_ADV_MAX_ADV_SET > 1) */
}
```

Reply:
Your suggestion here is not the intended behavior. In most we want to process the connection complete event immediately

Reply:
Right, my bad.